src/x86/intel_joule_expansion.c

```diff
@@ -145,6 +181,115 @@ mraa_joule_expansion_board()
     b->spi_bus[2].bus_id = 32766;
     b->spi_bus[2].slave_s = 1;
 
+    // Aio shield detection
+    // Only if i2c_designware.0 was successfully found
+    if (i2c_aio_bus != -1) {
+
+        syslog(LOG_NOTICE, "Attempting shield autodetection on I2C bus %d...", i2c_aio_bus);
+
+        i2c_smbus_ioctl_data_t sm;
+        i2c_smbus_data_t d;
+        int fp;
+
+        char i2c_fh[64];
+        snprintf(i2c_fh, 64, "/dev/i2c-%d", i2c_aio_bus);
+
+        sm.read_write = I2C_SMBUS_READ;
+        sm.command = 0x01;
+        sm.size = I2C_SMBUS_WORD_DATA; // LSB first
+        sm.data = &d;
+
+        if ((fp = open(i2c_fh, O_RDWR)) < 1)
```

Comment:
I2C raw would do exactly what's needed, however there isn't a single board definition that relies on the API headers. If that's something you encourage, I see no problem using the raw functions instead. It would also get rid of the smbus data structs since they would be pulled from MRAA's i2c.h...